### ISSUE TITLE
Irregular results if I try accessing https://readinglist.stage.mozaws.net vs http://readinglist.stage.mozaws.net:443

### ISSUE THREAD

**Issue by pdehaan** [mozilla-services/readinglist#210]
In the [./loadtests/Makefile:1](https://github.com/mozilla-services/readinglist/blob/b64cfb8bc7258e38dbd90cecc13b5d06ea3b3979/loadtests/Makefile#L1) file, we have this:
```ini
SERVER_URL = https://readinglist.stage.mozaws.net:443
```

If I run a small load test (`make test`), I get this:
```
.venv/bin/loads-runner \
  --config=./config/test.ini \
  --server-url=https://readinglist.stage.mozaws.net:443 \
  loadtest.TestBasic.test_all

Duration: 34.91 seconds
Hits: 849
Started: 2015-03-25 00:34:23.200250
Approximate Average RPS: 24
Average request time: 0.10s
Opened web sockets: 0
Bytes received via web sockets : 0

Success: 300
Errors: 0
Failures: 0
```

Removing the port and rerunning the tests gives us the same 300 successes:
```
.venv/bin/loads-runner \
  --config=./config/test.ini \
  --server-url=https://readinglist.stage.mozaws.net \
  loadtest.TestBasic.test_all

Duration: 41.94 seconds
Hits: 971
Started: 2015-03-25 00:35:49.526020
Approximate Average RPS: 23
Average request time: 0.11s
Opened web sockets: 0
Bytes received via web sockets : 0

Success: 300
Errors: 0
Failures: 0
```

If I change the `--server-url` from **https://**readinglist.stage.mozaws.net to **http://**readinglist.stage.mozaws.net**:443** then everything breaks and all 300 requests fail:
```
.venv/bin/loads-runner \
  --config=./config/test.ini \
  --server-url=http://readinglist.stage.mozaws.net:443 \
  loadtest.TestBasic.test_all

Duration: 16.68 seconds
Hits: 0
Started: 2015-03-25 00:39:13.474909
Approximate Average RPS: 0
Average request time: 0.00s
Opened web sockets: 0
Bytes received via web sockets : 0

Success: 0
Errors: 300
Failures: 0

1 occurrences of:
    ConnectionError: ('Connection aborted.', error(54, 'Connection reset by peer'))    Traceback:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 322, in run
    self.setUp()
  File "loadtest/__init__.py", line 53, in setUp
    self.create()
  File "loadtest/__init__.py", line 94, in create
    auth=self.basic_auth)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 504, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/loads/measure.py", line 80, in request
    method, url, headers=headers, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 461, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/loads/measure.py", line 88, in send
    res = _Session.send(self, request, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)

1 occurrences of:
    ConnectionError: ('Connection aborted.', error(54, 'Connection reset by peer'))    Traceback:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 322, in run
    self.setUp()
  File "loadtest/__init__.py", line 53, in setUp
    self.create()
  File "loadtest/__init__.py", line 94, in create
    auth=self.basic_auth)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 504, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/loads/measure.py", line 80, in request
    method, url, headers=headers, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 461, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/loads/measure.py", line 88, in send
    res = _Session.send(self, request, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)

1 occurrences of:
    ConnectionError: ('Connection aborted.', error(54, 'Connection reset by peer'))    Traceback:
  File "/System/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/unittest/case.py", line 322, in run
    self.setUp()
  File "loadtest/__init__.py", line 53, in setUp
    self.create()
  File "loadtest/__init__.py", line 94, in create
    auth=self.basic_auth)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 504, in post
    return self.request('POST', url, data=data, json=json, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/loads/measure.py", line 80, in request
    method, url, headers=headers, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 461, in request
    resp = self.send(prep, **send_kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/loads/measure.py", line 88, in send
    res = _Session.send(self, request, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/sessions.py", line 573, in send
    r = adapter.send(request, **kwargs)
  File "/Users/pdehaan/dev/github/readinglist/loadtests/.venv/lib/python2.7/site-packages/requests/adapters.py", line 415, in send
    raise ConnectionError(err, request=request)
```

---

Possibly just stupid user error, but I was a bit suspicious when I was running load tests and noticed that some of the slowest URLs were being returned without port numbers. I'm mildly worried that this may prevent us from being able to run a readinglist server on any arbitrary port.

| Stats by URLs (10 slowests): | Average request time: |
|----|----:|
| https://readinglist.stage.mozaws.net:443/v1/articles/c45c4e87-0abd-44ac-8f28-7a6bbdef4470                       	| 0.180247
| https://readinglist.stage.mozaws.net:443/v1/articles?_since=1427244414591                                       	| 0.158551
| https://readinglist.stage.mozaws.net:443/v1/articles/6c1cb422-445f-43cd-bb31-f1a1a4d9ee50                       	| 0.147060
| https://readinglist.stage.mozaws.net:443/v1/articles/21b4d13b-b5dc-4a32-907d-732df75b2566                       	| 0.142793
| **https://readinglist.stage.mozaws.net/v1/articles?_limit=20&_token=eyJsYXN0X21vZGlmaWVkIjogMTQyNzI0NDQxNzMxOX0%3D**	| 0.139831
| https://readinglist.stage.mozaws.net:443/v1/articles/e3fe363d-c0b3-4e7a-a634-0c48b72797ed                       	| 0.132846
| https://readinglist.stage.mozaws.net:443/v1/articles/7c02cd1c-fcf1-4217-b2bc-8c55e8e1f4e3                       	| 0.132594
| https://readinglist.stage.mozaws.net:443/v1/articles?_since=1427244418949                                       	| 0.129685
| https://readinglist.stage.mozaws.net:443/v1/articles/c0c27964-fb5a-4aae-a193-0f255e584bfa                       	| 0.129675
| **https://readinglist.stage.mozaws.net/v1/articles?_limit=20&_token=eyJsYXN0X21vZGlmaWVkIjogMTQyNzI0NDQxNTYyOH0%3D**	| 0.127290

**NOTE:** Note that the two results in **bold** above are missing the port number in the URL.

**Comment by leplatrem** [comment]
Thanks for reporting this!

Yesterday I fixed this bug here https://github.com/mozilla-services/cliquet/pull/148 but we did not release since it still has some errors on readinglist unit tests (like adding the port twice ``localhost:8000:8000``)

In shorts: we'll tacke this in a matter of minutes :)

**Comment by leplatrem** [comment]
Fixed in 1.3.0

**Comment by pdehaan** [comment]
@leplatrem Can you help confirm that this is fixed in 1.4.x?

I'm still seeing this in 1.4.0:

```sh
$ curl https://readinglist.stage.mozaws.net/v1/
```
```json
{
  "url": "https:\/\/readinglist.stage.mozaws.net\/v1\/",
  "documentation": "https:\/\/readinglist.readthedocs.org\/",
  "version": "1.4.0",
  "hello": "readinglist"
}
```

---

```
$ make test
.venv/bin/loads-runner --config=./config/test.ini --server-url=https://readinglist.stage.mozaws.net:443 loadtest.TestBasic.test_all
```

| Stats by URLs (10 slowests): | Average request time |
|----|----:|
| ~~https://readinglist.stage.mozaws.net/v1/articles?_limit=20&_token=eyJsYXN0X21vZGlmaWVkIjoxNDI3NDkxNzg2ODM3fQ%3D%3D~~ | 2.395968	
| ~~https://readinglist.stage.mozaws.net/v1/articles?_limit=20&_token=eyJsYXN0X21vZGlmaWVkIjoxNDI3NDkxNzgyNTg1fQ%3D%3D~~ | 1.121343
| https://readinglist.stage.mozaws.net:443/v1/articles?_since=1427491773391&deleted=true | 0.7857
| https://readinglist.stage.mozaws.net:443/v1/articles/45eff07a-d344-4d8a-bed3-b23cc21434ab | 0.666501
| https://readinglist.stage.mozaws.net:443/v1/articles/4db606a2-ffcd-4282-bb31-f1de2e297667 | 0.60943
| https://readinglist.stage.mozaws.net:443/v1/articles?_since=1427491815845 | 0.600214
| ~~https://readinglist.stage.mozaws.net/v1/articles?_limit=20&_token=eyJsYXN0X21vZGlmaWVkIjoxNDI3NDkxNzY2NjQ4fQ%3D%3D~~ | 0.5306055
| https://readinglist.stage.mozaws.net:443/v1/articles?_since=1427491788685 | 0.517616
| ~~https://readinglist.stage.mozaws.net/v1/articles?_limit=20&_token=eyJsYXN0X21vZGlmaWVkIjoxNDI3NDkxNzYxMjY1fQ%3D%3D~~ | 0.517107	
| https://readinglist.stage.mozaws.net:443/v1/articles/046a8213-8ca1-4e67-a43c-1c6c6c18d8f1 | 0.457185

**NOTE:** I did a ~~strikethrough~~ text on the URLs from the load test that were removing the specified :443 port. The good news is that all the URLs seem to be pagination links. I'll poke around a bit more with manual tests and see if this is a potential ReadingList issue or just a test issue.
